### ISSUE TITLE
Unexpected line breaks in longform posts

### ISSUE THREAD

**Issue by dmnyc** [damus-io/damus#1581]
I want to share how this kind 30023 long form post is displayed on Damus vs. Coracle. On Damus, there's some extra line breaks that shouldn't be there. Is there a reason for this, and can it be cleaned up somehow?

Source post from stacker.news:
https://stacker.news/items/274794

Reference note: 
note16vl9z038l3pgq8prya0zjx7wja3ume37ryzuagwga0ksetfchtdqj56yc6

nostr:nevent1qqsdx0j38cnlcs5qrs3jwh3fr08fwc7duclpjpww58ywhmgv45ut4kspz3mhxue69uhhyetvv9ujuerpd46hxtnfduq35amnwvaz7tmjv4kxz7fwdaexzmn8v4cxjmrv9ejx2asprpmhxue69uhhwetvvdhk6efwdehhxarj9emkjmn9qyv8wumn8ghj7un9d3shjtnwdaehgunpw35jucm0d5q3samnwvaz7tmsw43xc6tr9eex2mrp095kueewd9hsz9thwden5te0wfjkccte9ekk7um5wgh8qatzqyg8wumn8ghj7mn0wd68ytnhd9hx2qg0waehxw309ahzummtxqhx7un8qydhwumn8ghj7mn0wd68yetvv9uju7t9va58ymewwd5hgegpp4mhxue69uhkummn9ekx7mqnhuxpp

Screenshots:

![IMG_42772BDDCC24-1](https://github.com/damus-io/damus/assets/3638467/683e42ee-3e70-4fce-887d-a1fd12c73e87)
![Screenshot 2023-10-04 at 11 48 21 PM](https://github.com/damus-io/damus/assets/3638467/5b323cb9-b467-48c3-af2f-e347d658922c)

**Comment by dmnyc** [comment]
Update: According to Fabian, the line breaks were caused by hidden characters in the text, possibly carried over from a different editor. 

It should be posible to ignore these when rendering to produce a cleaner end result.

https://damus.io/note1gqq8v4grqlztv0y5k62dgcqjnqju84xwuvqsyhtz3nm986h2salq43t7nk

![image](https://github.com/damus-io/damus/assets/3638467/ff60b9c8-19b5-4ff7-9ef7-d451ef6f7543)

**Comment by alltheseas** [comment]
What app was used to create this post?

**Comment by dmnyc** [comment]
It was published in stacker.news but I wrote the text using Notes on iOS. 

When I analyzed the text in another editor I saw that it had inserted some invisible characeters coded as Unicode 0x2028 that appeared on screen as normal line breaks: 

![Screenshot 2023-10-05 at 10 37 32 AM](https://github.com/damus-io/damus/assets/3638467/8b0f9331-9cc4-4f8f-8115-de7ec6ebaa56)

**Comment by alltheseas** [comment]
Can you submit request to stacker news, as sn caused this issue

**Comment by dmnyc** [comment]
Yes, I can. But I am also curious as to how Coracle was able to render this post correctly while Damus and Nostur were not. Seems like something that could be parsed on the client side?

**Comment by jb55** [comment]
On Thu, Oct 05, 2023 at 08:18:09AM -0700, alltheseas wrote:
>> Yes, I can. But I am also curious as to how Coracle was able to
>> render this post correctly while Damus and Nostur were not. Seems
>> like something that could be parsed on the client side?
>
>@staab do you recall doing something to handle the issue described
>here?

It's likely the browser renderer just handles these properly
automatically. Both nostur and damus use ios renderers that are either
more dumb or accurate.

Ideally the tool that is broadcasting these notes could fix this up
before broadcast. Not sure if it makes sense to parse notes to filter
out bad linebreaks?

**Comment by dmnyc** [comment]
I've seen it from time to time and when I notice it, I always replace it with a standard line break. I guess I found a weird edge case!

**Comment by dmnyc** [comment]
The post looks perfect on habla.news BTW:
https://habla.news/a/naddr1qqsksar5wpen5te0wd6xzcmtv4ezumn9waej76t5v4khxtejxu6rwwf5qgswum4p82uluhz2dr40nvdrflspffntgqghc58w9fs57nx6jkdkuaqrqsqqqa28c5k9f3

**Comment by jb55** [comment]
we should confirm if your longform changes fixed this @alltheseas

**Comment by alltheseas** [comment]
investigating.

**Comment by alltheseas** [comment]
this should fix: https://github.com/damus-io/damus/pull/3659
